src/parser/monk/windwalker/modules/items/JadeIgnition.js

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import { Trans } from '@lingui/macro';
+import SPELLS from 'common/SPELLS';
+import SpellLink from 'common/SpellLink';
+import Statistic from 'interface/statistics/Statistic';
+import { STATISTIC_ORDER } from 'interface/others/StatisticBox';
+import STATISTIC_CATEGORY from 'interface/others/STATISTIC_CATEGORY';
+import { formatPercentage } from 'common/format';
+import AbilityTracker from 'parser/shared/modules/AbilityTracker';
+import BoringSpellValueText from 'interface/statistics/components/BoringSpellValueText';
+import ItemDamageDone from 'interface/ItemDamageDone';
+import Events from 'parser/core/Events';
+
+const MAX_STACKS = 30;
+
+class JadeIgnition extends Analyzer {
+  static dependencies = {
+    abilityTracker: AbilityTracker,
+  };
+  totalStacks = 0;
+  currentStacks = 0;
+  stacksWasted = 0;
+
+  constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.hasLegendaryByBonusID(SPELLS.JADE_IGNITION.bonusID);
+    if (!this.active) {
+      return;
+    }
+    this.addEventListener(Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.JADE_IGNITION_BUFF), this.applyBuff);
+    this.addEventListener(Events.applybuffstack.by(SELECTED_PLAYER).spell(SPELLS.JADE_IGNITION_BUFF), this.applyBuffStack);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.FISTS_OF_FURY_CAST), this.castFistsOfFury);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.SPINNING_CRANE_KICK), this.castSpinningCraneKick);
+  }
+
+  applyBuff() {
+    this.totalStacks += 1;
+    this.currentStacks += 1;
```

Comment:
If you're at max stacks you won't get an `applybuff` or `applybuffstack` event meaning you will have an inaccurate total stacks.
If someone never casted sck they could get a negative % value as stacks wasted would increase while total isn't.